src/main/java/uk/gov/hmcts/darts/common/datamanagement/api/DataManagementFacadeImpl.java

```diff
@@ -1,125 +1,173 @@
+        ObjectRecordStatusEntity storedStatus = objectRecordStatusRepository.getReferenceById(STORED.getId());
+        List<ExternalObjectDirectoryEntity> storedEodEntities = externalObjectDirectoryRepository.findByEntityAndStatus(transcriptionDocumentEntity,
+                                                                                                                        storedStatus);
+        if (CollectionUtils.isEmpty(storedEodEntities)) {
+            String errorMessage = MessageFormat.format("No eodEntities found for TranscriptionDocumentId {0,number,#}", transcriptionDocumentEntity.getId());
+            log.error(errorMessage);
+            throw new FileNotDownloadedException(errorMessage);
+        }
+        try {
+            return getDataFromStorage(storedEodEntities);
+        } catch (FileNotDownloadedException fnde) {
+            log.error("Could not retrieve file from any storage for TranscriptionDocumentId {}", transcriptionDocumentEntity.getId());
+            throw fnde;
+        }
     }
 
     @Override
-    public void getDataFromUnstructuredArmAndDetsBlobs(DownloadableExternalObjectDirectories downloadableExternalObjectDirectories) {
-
-        // process for all standard blob stores
-        downloadableExternalObjectDirectories.getEntities().forEach(dataToDownload -> CONTAINER_PROCESSING_ORDER.forEach(type -> {
-            Optional<BlobContainerDownloadable> container = getSupportedContainer(type);
-            if (container.isPresent() && !downloadableExternalObjectDirectories.getResponse().isProcessedByContainer()) {
-                Optional<String> containerName = container.get().getContainerName(type);
-                if (containerName.isPresent()
-                    && dataToDownload.isForLocationType(getForDatastoreContainerType(type))
-                    && processObjectDirectoryForContainerType(dataToDownload, type, configuration.isFetchFromDetsEnabled())) {
-                    log.info("Downloading blob id {} from container {}", dataToDownload.getId(), type.name());
-
-                    processDownloadResponse(type, dataToDownload, downloadableExternalObjectDirectories, container.get());
-                }
-            }
-        }));
+    public DownloadResponseMetaData retrieveFileFromStorage(AnnotationDocumentEntity annotationDocumentEntity) throws FileNotDownloadedException {
+        ObjectRecordStatusEntity storedStatus = objectRecordStatusRepository.getReferenceById(STORED.getId());
+        List<ExternalObjectDirectoryEntity> storedEodEntities = externalObjectDirectoryRepository.findByEntityAndStatus(annotationDocumentEntity, storedStatus);
+        if (CollectionUtils.isEmpty(storedEodEntities)) {
+            String errorMessage = MessageFormat.format("No eodEntities found for AnnotationDocId {0,number,#}", annotationDocumentEntity.getId());
```

Comment:
Prefer something similar to mediaId, so `annotationDocumentId` here and probably `transcriptionDocumentId` in the other place I just seen to try to keep to similar styles in the Facade methods